### PR TITLE
feat(www): features page revamp

### PR DESCRIPTION
**Pull request by gillkyle**
## Description

This is a revamp of the Features page. New technologies are compared and reevaluated scores have been estimated to along with each (though many are surely not totally up to date, there is an issue tracking comments about scores at #16233).

There are now pages generated for each combination of comparisons between Gatsby and a set of technologies. A couple of shortcut links to what I thought would be common comparisons are included in the features sidebar. 

Appreciate any feedback. After speaking with @prestonso the idea was to ship what we have for this and update any scores as people give more feedback. ~The only data that is not up to date in one of the csv files (that we need to complete before merging) is the JAMstack frameworks column in the features overview, everything else should be accurate based on the evaluations~ everything is now up to date based on the info in this sheet: https://docs.google.com/document/d/1z5-qltGRRX1ClYkQpmM883DTjbAc7SDatOeAwilK42E/edit#heading=h.dmqmonp7qtox

## Screenshots

<details>
<summary>Overview</summary>

localhost_8000_features_

</details>

<details>
<summary>JAMstack</summary>

localhost_8001_features_ (1)

</details>

<details>
<summary>Comparison Page</summary>

features-fin

</details>

### CODE REVIEW

File: www/src/data/features/comparison-options.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      "key": "hugo",
+      "display": "Hugo",
+      "nodeField": "Hugo",
+      "description": "Hugo is a static site generator written in Go. Hugo takes data files, i18n bundles, configuration, templates for layouts, static files, and content written in Markdown and renders a static website. It is an open source project licensed under the Apache License 2.0."
+    },
+    {
+      "key": "nuxtjs",
+      "display": "Nuxt.js",
+      "nodeField": "Nuxtjs",
+      "description": "Next.js is a free and open source web application framework based on Vue.js, Node.js, webpack and Babel.js for building server-side rendered and/or static web applications using Vue."
```
Comment (LekoArts) [inline www/src/data/features/comparison-options.json]: Typo here

Comment (DSchau) [reply www/src/data/features/comparison-options.json]:
```suggestion
      "description": "Nuxt.js is a free and open source web application framework based on Vue.js, Node.js, webpack and Babel.js for building server-side rendered and/or static web applications using Vue."
```

Comment (LekoArts) [reply www/src/data/features/comparison-options.json]: Thanks! GitHub on mobile is just inferior

File: www/src/utils/get-csv-features-data.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export const getFeaturesData = function(data) {
```
[the remark this answers is not in the document]
Comment (gillkyle) [reply www/src/utils/get-csv-features-data.js]: I added a very rudimentary test, haven't written a lot of tests in my day, so would appreciate any critiques!

**Pull request merged**